### PR TITLE
fix: refactor the node generation to break out interface of list resources

### PR DESCRIPTION
**Pull request by childish-sambino**

### CODE REVIEW

File: examples/node/lib/rest/api/v2010/accounts/calls/feedbackSummary.ts [hunk 1]
```diff
@@ -20,32 +21,49 @@ import V2010 from '../../../V2010';
...
-    constructor(protected _version: V2010, accountSid: string) {
-        this._solution = { accountSid };
-        this._uri = `/2010-04-01/Accounts/${accountSid}/Calls/FeedbackSummary.json`;
-    }
-
     /**
      * Create a FeedbackSummarInstance
      *
-     * @param { FeedbackSummarInstanceCreateOptions } params - Parameter for request
+     * @param { FeedbackSummarListInstanceInstanceCreateOptions } params - Parameter for request
```
Comment (beebzz) [inline examples/node/lib/rest/api/v2010/accounts/calls/feedbackSummary.ts:35]: Not sure if `ListInstanceInstance` is the best naming convention. Also, is `FeedbackSummar` a typo?

Comment (childish-sambino) [reply examples/node/lib/rest/api/v2010/accounts/calls/feedbackSummary.ts]: Fixed the duplicate `Instance` naming.

The missing `y`  on summary is it trying to singularize the resource name. There's a ticket to fix this.

File: src/main/resources/twilio-node/operation.mustache [hunk 2]
```diff
@@ -0,0 +1,61 @@
+{{#hasParams}}
+    {{#hasRequiredParams}}
+        if (params === null || params === undefined) {
+            throw new Error('Required parameter "params" missing.');
+        }
+
+        {{#requiredParams}}
+        {{^isBodyParam}}
```
Comment (rakatyal) [inline src/main/resources/twilio-node/operation.mustache:8]: Where are we passing this parameter(isBodyParam) to the mustache file?

**Review by rakatyal — APPROVED** [review]

File: src/main/java/com/twilio/oai/TwilioNodeGenerator.java [hunk 3]
```diff
@@ -48,7 +48,6 @@ public void processOpts() {
         additionalProperties.put("domain", StringUtils.camelize(domain));
 
         supportingFiles.clear();
-        supportingFiles.add(new SupportingFile("tsconfig.mustache", "tsconfig.json"));
```
Comment (rakatyal) [inline src/main/java/com/twilio/oai/TwilioNodeGenerator.java:51]: Nit: Any compilation impact on deleting tsconfig.json?

File: src/main/resources/twilio-node/operation.mustache (hunk 2 above)
Comment (childish-sambino) [reply src/main/resources/twilio-node/operation.mustache:8]: This is done by the base openapi-generator which is populating the properties. No customization is needed.

File: src/main/java/com/twilio/oai/TwilioNodeGenerator.java (hunk 3 above)
Comment (childish-sambino) [reply src/main/java/com/twilio/oai/TwilioNodeGenerator.java:51]: Nope. We should have one defined at the repo-root so these aren't necessary.

**Pull request merged**